packages/webcomponents/src/components/business-forms/owner-form/owner-form-core.tsx

```diff
@@ -0,0 +1,227 @@
+import { Component, Host, h, Prop, State, Event, EventEmitter, Method, Watch } from '@stencil/core';
+import { FormController } from '../../form/form';
+import { Identity, Owner } from '../../../api/Identity';
+import { BusinessFormClickActions, BusinessFormClickEvent, BusinessFormSubmitEvent } from '../utils/business-form-types';
+import { ComponentError } from '../../../api/ComponentError';
+import { identitySchema } from '../schemas/business-identity-schema';
+import { parseIdentityInfo } from '../utils/payload-parsers';
+
+@Component({
+  tag: 'owner-form-core'
+})
+export class BusinessOwnerFormCore {
+  @State() isLoading: boolean = false;
+  @State() formController: FormController;
+  @State() errors: any = {};
+  @State() owner: Owner = {};
+
+  @Prop() authToken: string;
+  @Prop() ownerId?: string;
+  @Prop() businessId?: string;
+  @Prop() getOwner: Function;
+  @Prop() patchOwner: Function;
+  @Prop() postOwner: Function;
+  @Prop() allowOptionalFields?: boolean;
+  @Prop() removeOwner: (id: string) => void;
+  @Prop() newFormOpen?: boolean;
+  @Prop() ownersLength?: number;
+
+  @Watch('isLoading')
+  loadingWatcher() {
+    this.formLoading.emit(this.isLoading);
+  }
+
+  @Event({ bubbles: true }) submitted: EventEmitter<BusinessFormSubmitEvent>;
+  @Event({ bubbles: true }) ownerSubmitted: EventEmitter<any>;
+  @Event({ eventName: 'click-event', bubbles: true }) clickEvent: EventEmitter<BusinessFormClickEvent>;
+  @Event() formLoading: EventEmitter<boolean>;
+  @Event({ eventName: 'error-event', bubbles: true }) errorEvent: EventEmitter<ComponentError>;
+
+  constructor() {
+    this.validate = this.validate.bind(this);
```

Comment:
I think this part is not needed. I removed the entire constructor locally and the form still worked. Can you confirm this?

Reply:
Hey finally getting back to this - 

I think the constructor is needed. Because of the way that this form step handles refs for the various forms that are rendered - if we remove this constructor then clicking on the form specific buttons will return a Type error 

I'd love to use arrow syntax on the methods to bind them to the component that way instead, but unfortunately you cannot write stencil `Method` functions using that syntax. 

If you can think of a way around it, lemme know! Otherwise I think this will need to stay for now. 

![Screenshot 2024-09-10 at 6 46 36 PM](https://github.com/user-attachments/assets/c5ffda27-9f9e-4c2b-bb7a-005965902500)

Reply:
Okay, sounds good then!